_posts/2024-01-30-securing-wildfly-apps-oidc-k8s.adoc

```diff
@@ -0,0 +1,398 @@
+---
+layout: post
+title: 'Securing WildFly Apps with OIDC on Kubernetes'
+date: 2024-01-30
+tags: oidc kubernetes keycloak adapter galleon-pack
+synopsis: Learn how to secure applications deployed to WildFly on Kubernetes with OIDC.
+author: theashiot
+---
+
+:toc: macro
+:toc-title:
+
+You can secure your WildFly applications deployed on Kubernetes with OpenID Connect (OIDC). By using OIDC to secure applications, you delegate authentication to OpenID providers. This guide shows how to secure an example application deployed to WildFly on a Kubernetes cluster running on your local machine, with OIDC using Keycloak as the OpenID provider.
+
+toc::[]
+
+== Prerequisites
+
+To follow along with this guide, you will need:
+
+* Roughly 15 minutes.
+* https://docs.docker.com/engine/install/[Docker] or https://podman.io/docs/installation[Podman].
+* A Kubernetes cluster.
++
+You can use a local Kubernetes cluster such as link:https://minikube.sigs.k8s.io/docs/[minikube], or https://kind.sigs.k8s.io/[kind].
++
+The steps provided in this guide use minikube.
+
+* Kubernetes command-line tool, https://kubernetes.io/docs/tasks/tools/[kubectl].
+
+* A public container registry to host your application's Docker image.
++
+For example, https://hub.docker.com/[Docker
+Hub].
+* https://helm.sh/docs/intro/install/[Helm Chart].
+
+
+== Example Application
+
+We use a simple web application in this guide that consists of a single https://github.com/wildfly-security-incubator/elytron-examples/blob/main/simple-webapp-oidc/src/main/java/org/wildfly/security/examples/SecuredServlet.java[servlet]. We show how to secure this servlet using OIDC. We will use the example in the https://github.com/wildfly-security-incubator/elytron-examples/tree/main/simple-webapp-oidc[simple-webapp-oidc] directory in the https://github.com/wildfly-security-incubator/elytron-examples[elytron-examples] repository.
+
+To obtain this example, clone the elytron-examples repository to your local machine:
+
+[source]
+----
+git clone git@github.com:wildfly-security-incubator/elytron-examples.git
+----
+
+== Start Keycloak
+
+We will be using Keycloak as our OpenID provider.
+
+Follow the instructions, up until "Log in to the Admin Console", provided in the https://www.keycloak.org/getting-started/getting-started-kube[Get started with Keycloak on Kubernetes] guide.
+
+== Configure Keycloak
+
+To create a realm in Keycloak, add a user and role to the realm, and create a client with which to secure your application, follow these steps:
+
+. Log into the Keycloak Admin Console using the username and password you specified earlier.
+
+. Create a new realm called *myrealm*. For more information, see the Keycloak documentation on how to https://www.keycloak.org/getting-started/getting-started-kube#_create_a_realm[create a realm].
```

Comment:
The directions linked to are old and not that helpful.  Use this link and
instruct the user to Follow the directions in "Creating a realm".  
https://www.keycloak.org/docs/latest/server_admin/#proc-creating-a-realm_server_administration_guide

Reply:
upated